### ISSUE TITLE
Assumption that passwords will be the same across instances when using indexer clustering

### ISSUE THREAD

**Issue by pgullette** [splunk/splunk-ansible#206]
I'm working a distributed deployment, and in my case I was starting with the Splunk AMI for AWS. This AMI has splunk-ansible already installed which is great and saves some steps, but the behavior of using that AMI is to create the admin password for splunk based on the created instance id.

I modified the playbook to copy instance-specific default.yml files. Within each default.yml, I can define the unique parts such as password. I noticed when I did my deploy including an indexer cluster that it was failing during peering the search head to the cluster master. Specifically, in peer_cluster_master.yml, there is an include to wait for the cluster master instance. That task is using the in-scope splunk user and password, which in this case is the admin user and password for the search head. This obviously creates a 401: Unauthorized when the admin password on the CM is different than SH.

There are ways to solve this, but I was curious if this was an assumption about using this for a distributed setup.

**Comment by nwang92** [comment]
Yeah at the moment we do assume common passwords across all instances of your Splunk Enterprise deployment. We're currently investigating ways to open up the ability to define usernames/passwords on a per-instance or per-role basis.

As far as the Splunk AMI for AWS goes, I imagine the possible workarounds would be:
* After instance launch, change the admin password from instance-id to something else before running ansible
* Configure user-defined password at initial launch? I'm not sure if this is possible, but I imagine this would be a nice feature (cc @rcabral-splunk)

It seems to me that if every Splunk EC2 gets a dynamic password (based on instance-id) without the ability to pre-define it, you'll likely never be able to use the Splunk AMI for AWS for any distributed environments. Not sure if that's by design, but assuming you are able to get a distributed environment brought up, it will be very brittle and error-prone when instances die and get recreated.

**Comment by bb03** [comment]
I tried this with the docker-splunk plays.  Speciicially I took the play here: https://github.com/splunk/docker-splunk/blob/develop/test_scenarios/3idx3sh1cm.yaml and modified it so that each tier contained different passwords (i.e. ```SPLUNK_PASSWORD=${SPLUNK_PASSWORD_IDX}``` for the indexer environment variables, similar thing for the search head and cluster master).  A little surprisingly, the splunk environment I got back appears to be correct, even though I am seeing this error in my logs
```idx2    | TASK [splunk_indexer : Check Splunk instance is running] ***********************
idx2    | fatal: [localhost]: FAILED! => {
idx2    |     "attempts": 60, 
idx2    |     "cache_control": "private", 
idx2    |     "changed": false, 
idx2    |     "connection": "Close", 
idx2    |     "content": "{\"messages\":[{\"type\":\"ERROR\",\"text\":\"Unauthorized\"}]}", 
idx2    |     "content_length": "53", 
idx2    |     "content_type": "application/json; charset=UTF-8", 
idx2    |     "date": "Fri, 10 Jan 2020 00:04:28 GMT", 
idx2    |     "elapsed": 0, 
idx2    |     "json": {
idx2    |         "messages": [
idx2    |             {
idx2    |                 "text": "Unauthorized", 
idx2    |                 "type": "ERROR"
idx2    |             }
idx2    |         ]
idx2    |     }, 
idx2    |     "redirected": false, 
idx2    |     "server": "Splunkd", 
idx2    |     "status": 401, 
idx2    |     "url": "https://cm1:8089/services/server/info?output_mode=json", 
idx2    |     "vary": "Cookie, Authorization", 
idx2    |     "www_authenticate": "Basic realm=\"/splunk\"", 
idx2    |     "x_content_type_options": "nosniff", 
idx2    |     "x_frame_options": "SAMEORIGIN"
idx2    | }
idx2    | 
idx2    | MSG:
idx2    | 
idx2    | Status code was 401 and not [200]: HTTP Error 401: Unauthorized
idx2    | 
idx2    | ...ignoring```
Which - given the issue - isn't that shocking because its trying to do exactly what this ticket describes - log into the cluster master with the indexer password.

I know the license master is a separate issue, I just wanted to see the extent of the problem and how much will need to be changed to support this.
